homeassistant/components/device_tracker/keenetic_ndms2.py

```diff
@@ -69,53 +72,32 @@ def scan_devices(self):
 
     def get_device_name(self, device):
         """Return the name of the given device or None if we don't know."""
-        filter_named = [result.name for result in self.last_results
-                        if result.mac == device]
-
-        if filter_named:
-            return filter_named[0]
-        return None
+        name = next((
+            result.name for result in self.last_results
+            if result.mac == device), None)
+        return name
+
+    def get_extra_attributes(self, device):
+        """Return the IP of the given device."""
+        attributes = next((
+            {'ip': result.ip} for result in self.last_results
```

Comment:
Are we using `'ip'` elsewhere as attribute name? I'd prefer to use `'ip_address'`.

Reply:
Yes, we are: `homeassistant.components.device_tracker.nmap_tracker.NmapDeviceScanner#get_extra_attributes`
There is also `homeassistant.components.device_tracker.unifi.UnifiScanner#get_extra_attributes` but the contents of this dict are not known from the sigratures - this component passes JSON response from API with no changes and explanations

Reply:
Ok let's keep it as is then.